# CSCI 5481

## Homework 1
## Christopher White
## September 9, 2018

---
### PRINT PYTHON FILE


In [1]:
%cat exercise01.py

# Output file for CSCI 5481 Fall 2018 Exercise 1
# Christopher White
# September 19, 2018
#
# Usage:
# exercise01.py -h
import sys, os
import argparse
from subprocess import Popen, PIPE

def make_arg_parser():
    parser = argparse.ArgumentParser(prog='exercise01.py',
                          # version="%prog 1.0",
                          formatter_class=argparse.ArgumentDefaultsHelpFormatter)
    parser.add_argument("-q","--query",
                      default=argparse.SUPPRESS,
                      required=True,
                      help="Path to query fasta [required]") 
    parser.add_argument("-r","--ref",
                      default=argparse.SUPPRESS,
                      required=True,
                      help="Path to reference fasta [required]")
    parser.add_argument("-t","--taxonomy",
                      default=None,
                      required=True,
                      help="Path to taxonomy file [required]")
    parser.add_argument("-o","--output",
   

### RUN COMMAND LINE

In [2]:
!python exercise01.py -q query.fna -r ref.fna -t taxonomy.txt -o output.txt -c ./burst -V

./burst -q query.fna -r ref.fna -t taxonomy.txt -o output.txt

Return Value: 0

STDOUT: ...
 --> Setting threads to 0
Using up to AVX-128 with 0 threads.
Parsed 130727 queries.
Max query len: 101, avg. divergence: 90.764586 (18.509566 w/o dupes)
Parsed 5000 references.
There are 5000 references and hence 312 clumps (+1)
Average R pack length = 1423.881789
Searching best paths through 35294 unique queries...

Search Progress: [0.32%]
Search Progress: [0.64%]
Search Progress: [0.96%]
Search Progress: [1.28%]
Search Progress: [1.60%]
Search Progress: [1.92%]
Search Progress: [2.24%]
Search Progress: [2.56%]
Search Progress: [2.88%]
Search Progress: [3.19%]
Search Progress: [3.51%]
Search Progress: [3.83%]
Search Progress: [4.15%]
Search Progress: [4.47%]
Search Progress: [4.79%]
Search Progress: [5.11%]
Search Progress: [5.43%]
Search Progress: [5.75%]
Search Progress: [6.07%]
Search Progress: [6.39%]
Search Progress: [6.71%]
Search Progress: [7.03%]
Search Progress: [7.35%]
Search Progre

---
## PROCESS OUTPUT FILE


In [3]:
import pandas as pd
import numpy as np

In [4]:
df = pd.read_csv('output.txt', header=None, delim_whitespace=True,
                names=['qseqid','sseqid','pident','length','mismatch','gapopen','qstart','qend','sstart','send','evalue','bitscore'])

In [5]:
df.head(5)

qseqid  sseqid     pident  length  mismatch  gapopen  \
0     h146M.1.418838_256197075  851865  97.029701     101         3        0   
1     h165M.1.418394_282943524  851865  97.029701     101         3        0   
2  Amz5chldF2.418405_806016025  851865  97.826088      92         2        0   
3  Amz4chldM2.418774_769559065  851865  97.029701     101         3        0   
4     h257M.1.418454_646157849  851865  96.969696      99         3        0   

   qstart  qend  sstart  send  evalue  bitscore  
0       1   101     308   409       3         0  
1       1   101     308   409       3         1  
2       1    92     308   400       2         0  
3       1   101     308   409       3         0  
4       1    99     308   407       3         0

### Question 1: What fraction of the input query sequences had a match in the database at 97% or above?

In [6]:
df[df['pident']>=97].count()['qseqid'], df['pident'].count()

(45550, 46011)

There are 45,550 at 97% or above out of 46,011 entries.

### Question 2: What is the most common bacterial species in the query set?

In [7]:
df['qseqid'].value_counts()

k278A.2.418424_651300242          1
h9M.1.418588_808479870            1
Amz6chldM.418668_793073478        1
TS109.418691_885856118            1
Amz6teen.418569_1029400378        1
TS129.418618_381258008            1
USchp33ChildB.418578_213517173    1
TS7.418860_392489734              1
TS195.418848_408859884            1
h95M.1.418831_743636910           1
TS129.418618_398711882            1
h264M.1.418377_251539154          1
h147M.1.418531_262271267          1
TS111.418684_365510845            1
USchp33ChildB.418578_234887031    1
h9M.1.418588_777264590            1
Amz29adlt.418370_1021060745       1
k278A.2.418424_647844107          1
USchp25Child.418345_206657105     1
TS109.418691_921630454            1
Amz29adlt.418370_1017418414       1
TS111.418684_379500016            1
TS1.418828_346501103              1
USchp18Mom.418783_228226310       1
Amz6chldM.418668_767329927        1
TS1.418828_355543881              1
h273M.1.418507_296460623          1
h68M.1.418773_188195634     

The answer is that all of the id's are unique.

### Question 3: What is the average percent similarity of the matches?

In [8]:
np.average(df['pident'])

98.46259236202215

The average of the percent similarity is 98.4626%.